docs/internals/security.txt

```diff
@@ -26,6 +26,15 @@ implications, please send a description of the issue via email to
 ``security@djangoproject.com``. Mail sent to that address reaches the `security
 team <https://www.djangoproject.com/foundation/teams/#security-team>`_.
 
+.. note::
+
+   When reporting a potential security vulnerability, it is important that
+   your code follows security best practices. Users have a responsibility to
+   ensure their code adheres to general standards, as Django cannot mitigate
+   vulnerabilities introduced by user code. A common example is the failure
+   to sanitize user input. For other examples, see the :ref:`security
+   topics <cross-site-scripting>`.
+
```

Comment:
Hm I had to read this a few times to understand it. Perhaps this suggestion is more comprehensible?

```suggestion
   Before reporting a potential security vulnerability, it is important to ensure that
   your own code follows security best practices, in order to allow maintainers the best course of action. Users have a responsibility to
   ensure their code adheres to general standards, as Django cannot mitigate
   vulnerabilities introduced by user code. A common example is the failure
   to sanitize user input. For other examples, see the :ref:`security
   topics <cross-site-scripting>`.
```

I'm not part of the security team so the "in order to allow maintainers the best course of action" part will need consideration for tweaking/removal - but the general gist here is to explicitly state the purpose of this paragraph 👍

Reply:
I like this edit, yes, it’s good to put in some reasoning. The only edit I’d make is to change “your own code” to “your code”, as “own” is redundant.

Reply:
How about something like the following, being just above the "Sending encrypted reports" admonition
```
.. admonition:: Example exploits must adhere to security standards

   Ensure your code follows security best practices before reporting a potential
   vulnerability or exploit. Developers have a responsibility to adhere to
   security standards, such as sanitizing user input. For other examples, see
   :doc:`security in Django </topics/security>`.

   Django will not mitigate vulnerabilities introduced by user code. That
   artificial intelligence can generate insecure code has no impact on developer
   responsibility, and hence requires no action from the security team.
```
I think we specifically wanted to call out code generated by ai as an argument of "even chatGPT is doing this" isn't a valid argument. I also updated "cannot" to "will not" as we chose to trust people will/should do things like sanitize inputs.

Reply:
On the reasoning, I guess it is "to enable a smooth investigation" or something like this
Essentially, if a demonstration of a security exploit doesn't follow best practices it is mostly invalid. It _might_ not be invalid and just be a bad example - if that's the case, it's distracting from the "real" issue

Reply:
Yup 👍 

Just to be clear though about my main confusion was what point the paragraph was trying to make 😅 then I realised it was basically "make sure your own code is secure before submitting stuff" … keyword "before" (and I still maintain "own" isn't redundant as in not "your code that's using Django" rather "the code you yourself have written should be secure")

Reply:
> How about something like the following, being just above the "Sending encrypted reports" admonition
> 
> ```
> .. admonition:: Example exploits must adhere to security standards
> 
>    Ensure your code follows security best practices before reporting a potential
>    vulnerability or exploit. Developers have a responsibility to adhere to
>    security standards, such as sanitizing user input. For other examples, see
>    :doc:`security in Django </topics/security>`.
> 
>    Django will not mitigate vulnerabilities introduced by user code. That
>    artificial intelligence can generate insecure code has no impact on developer
>    responsibility, and hence requires no action from the security team.
> ```
> 
> I think we specifically wanted to call out code generated by ai as an argument of "even chatGPT is doing this" isn't a valid argument. I also updated "cannot" to "will not" as we chose to trust people will/should do things like sanitize inputs.

i concor with adding it above the "Sending encrypted reports" admonition